### ISSUE TITLE
integrate ttop into ddc

### ISSUE THREAD

**Issue by rsvihladremio** [dremio/dremio-diagnostic-collector#31]
it would be _massively_ useful to include ttop in ddc. we should embed the binary so as to simplify distribution

- [ ] add license header along with the apache license and the files affected (in this case just sjk-tool.jar) into a NOTICE-bin file use this example https://github.com/apache/bookkeeper/blob/master/bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt
- [ ] make a folder called "lib" at the top level of the project
- [ ] add the jar to the folder
- [ ] add the "lib" folder to .gitignore ie this entry`lib/`
- [ ] update ./script/build to embed the jar into a file using go:embed https://pkg.go.dev/embed
- [ ] add a collect function in cmd/local/ttopcollect/ that uses that file to write out the binary and execute it using java `java -jar sjk-tools.jar -ri 1 -n 100 -p %v` the -p being for pid
- [ ] call function in /cmd/local/local.go and run it on another thread
- [ ] update the./cmd/local/conf/defaults.go to add a flag for capture time with the defaultCaptureSeconds
- [ ] update the./cmd/local/conf/defaults.go to add a flag for collecting ttop with a default of true
- [ ] update ./cmd/local/conf/conf_key_names.go and add the flags keys (something like --ttop-collection-seconds, and --collect-ttop )

**Comment by markcurtis1970** [comment]
ttop gives us several ways to collect (and sort)

```    -o, --order
       Sort order. Value tags: CPU, USER, SYS, ALLOC, NAME
       Default: [CPU]
```

I usually would get both CPU and ALLOC ?

**Comment by rsvihladremio** [comment]
I think we have too many flags already, but I would take a PR on this assuming adequate testing on the permutations that result (need to test all the different sort orders, what do you do with a bad sort order), as well as thead count and all the ttop options documented, however, it's definitely out of scope for this issue as it may take longer than the work so far.

Further more as far as usefullness:

1. sort can be changed by any aggregation tool consuming the threads
2. most users rarely read straight ttop and rely on some tool for analysis and aggregation, this is just because there are a lot of threads and one may need to aggregate them to see globally how much a multithreaded process or phase is using
3. there is rarely a case with _very_ _low_ cpu utilization and very high gc allocation (I would say never?) so an adequate thread count in an analysis report will allow one to get the biggest users of cpu and allocation and then.
4. raising thread count too high (-n option in ttop) increases collection time and cost, likewise it causes pain on the post analysis end.

TLDR I think 100 threads with a default sort gets us nearly all the way there
